docs/configuration.md

```diff
@@ -0,0 +1,85 @@
+<!--
+    Copyright 2019 VMware, Inc.
+    SPDX-License-Identifier: BSD-2-Clause
+-->
+
+# Precaution configuration
+
+Precaution is using a variety of tools under the hood without exposing
+implementation details and complexity of the underlying tools to the user.
+
+Concretely, this means that Precaution ignores any tool specific configuration
+files which may exist in the project's repository and performs scans based only
+on Precaution specific configuration.
+
+Precaution provides you with the option to use optional `.precaution.yaml` file.
+This file **must** be in the root directory of your project.
+
+All paths in the options should be **absolute**.
+
+This file can be configured with the following options:
+
+| Option | Syntax |
+|---|---|
+| [Exclude](#exclude) | [Globs](#globs) |
+
+## Exclude
+
+This option can be used to tell Precaution to exclude files or folders from processing.
+
+If you want to ignore `src/cmd/main.go` file you will add this to your configuration file:
+```
+exclude:
+- "src/cmd/main.go"
+```
+
+Another use-case is when you are using Golang and your dependencies are in
+the `vendor/` folder, then you can tell Precaution to skip all files
+within the `vendor/` folder like this:
+```
+exclude:
+- "vendor/**"
+```
+
+It's possible that you wouldn't want Precaution to scan your Golang test files and if the suffix in their names is `_test.go`
+then you can easily exclude them:
+```
+exclude:
+- "*_test.go"
+```
+
+The exclude option in the `.precaution.yaml` file supports multiple
+exclude rules:
+```
+exclude:
+- "*_test.go"
+- "vendor/**"
+- "src/cmd/main.go"
+```
+
+## Globs
```

Comment:
It feels a bit awkward going straight from a discussion of the exclude option to some documentation of globs without tying the two together. Perhaps it would make more sense to have the globs section as a sub-section of the Exclude section (at least whilst it's the only option)?

```suggestion
### Globs


Rules specified in the exclude option support globbing to match patterns of files, rather than having to hard-code a list. The following table defines pattern formats supported by Precaution configuration files:
```

Reply:
I thought that we will reference the `Globs` section from this table:
![image](https://user-images.githubusercontent.com/16246778/57948310-fa167680-78e9-11e9-9dc3-855ee7c708ff.png)
which makes it not a context of the `exclude` options section.

Reply:
That's the only part of the document that ties the two together, nowhere else is it made explicit that the Exclude option supports Glob syntax. With the current layout it's easy to overlook, thus I'd like us to be more explicit and tie the two sections together more explicitly somehow. Even if that's just mentioning in the Exclude section that the rules support glob syntax.

Reply:
I will add a line explaining that in the exclude section itself.
That way if we reference that section it will be easily visible what syntax it uses.